core/src/main/java/org/apache/accumulo/core/util/compaction/CompactionServicesConfig.java

```diff
@@ -0,0 +1,186 @@
+      return aconf.getAsBytes(Property.TSERV_MAJC_THROUGHPUT);
+    }
+
+    return ConfigurationTypeHelper
+        .getMemoryAsBytes(Property.TSERV_COMPACTION_SERVICE_DEFAULT_RATE_LIMIT.getDefaultValue());
+  }
+
+  @SuppressWarnings("removal")
+  private Map<String,String> getConfiguration(AccumuloConfiguration aconf) {
+
+    Map<String,String> configs =
+        aconf.getAllPropertiesWithPrefix(Property.TSERV_COMPACTION_SERVICE_PREFIX);
+
+    // check if deprecated properties for compaction executor are set
+    if (aconf.isPropertySet(Property.TSERV_MAJC_MAXCONCURRENT, true)) {
+
+      String defaultServicePrefix =
+          Property.TSERV_COMPACTION_SERVICE_PREFIX.getKey() + DEFAULT_SERVICE.canonical() + ".";
+
+      // check if any properties for the default compaction service are set
+      boolean defaultServicePropsSet = configs.keySet().stream()
+          .filter(key -> key.startsWith(defaultServicePrefix)).map(Property::getPropertyByKey)
+          .anyMatch(prop -> prop == null || aconf.isPropertySet(prop, true));
+
+      if (defaultServicePropsSet) {
+
+        String warning = String.format(
+            "The deprecated property %s was set. Properties with the prefix %s "
+                + "were also set, which replace the deprecated properties. The deprecated "
+                + "property was therefore ignored.",
+            Property.TSERV_MAJC_MAXCONCURRENT.getKey(), defaultServicePrefix);
+
+        deprecationWarningConsumer.accept(warning);
+
+      } else {
+        String numThreads = aconf.get(Property.TSERV_MAJC_MAXCONCURRENT);
+
+        // Its possible a user has configured the other compaction services, but not the default
+        // service. In this case want to produce a config with the default service configs
+        // overridden using deprecated configs.
+
+        HashMap<String,String> configsCopy = new HashMap<>(configs);
+
+        Map<String,String> defaultServiceConfigs =
+            Map.of(defaultServicePrefix + "planner", DefaultCompactionPlanner.class.getName(),
+                defaultServicePrefix + "planner.opts.executors",
+                "[{'name':'deprecated', 'numThreads':" + numThreads + "}]");
+
+        configsCopy.putAll(defaultServiceConfigs);
+
+        String warning = String.format(
+            "The deprecated property %s was set. Properties with the prefix %s "
+                + "were not set, these should replace the deprecated properties. The old "
+                + "properties were automatically mapped to the new properties in process "
+                + "creating : %s.",
+            Property.TSERV_MAJC_MAXCONCURRENT.getKey(), defaultServicePrefix,
+            defaultServiceConfigs);
+
+        deprecationWarningConsumer.accept(warning);
+
+        configs = Map.copyOf(configsCopy);
+      }
+    }
+
+    return configs;
+
+  }
+
+  public CompactionServicesConfig(AccumuloConfiguration aconf,
+      Consumer<String> deprecationWarningConsumer) {
```

Comment:
Passing a consumer for the logger is nifty but I think this could be confusing for someone debugging looking at line numbers for a logger. Also, if you just passed the logger, you could use it for debug logging too.

Reply:
Yea good point. A consumer is used here so that we could ignore duplicate logs coming from CompationManager. See `CompactionManager.warnAboutDeprecation()`. I can't think of a way to preserve this functionality with just a logger being passed.

Reply:
Ah OK I see now. That is a nice feature so its worth using the consumer to prevent spamming the logs.